packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/diagramfilter/PinButtonDescriptionProvider.java

```diff
@@ -0,0 +1,75 @@
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.diagrams.Diagram;
+import org.eclipse.sirius.components.forms.ButtonStyle;
+import org.eclipse.sirius.components.forms.WidgetIdProvider;
+import org.eclipse.sirius.components.forms.description.ButtonDescription;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.web.services.diagramfilter.api.IDiagramFilterActionContributionProvider;
+import org.springframework.stereotype.Service;
+
+/**
+ * Provides the description of the pin button for the diagram filter's split button.
+ *
+ * @author gdaniel
+ */
+@Service
+public class PinButtonDescriptionProvider implements IDiagramFilterActionContributionProvider {
+
+    private IObjectService objectService;
+
+    private DiagramFilterHelper diagramFilterHelper;
+
+    public PinButtonDescriptionProvider(IObjectService objectService, DiagramFilterHelper diagramFilterHelper) {
+        this.objectService = objectService;
+        this.diagramFilterHelper = diagramFilterHelper;
+    }
+
+    @Override
+    public ButtonDescription getButtonDescription() {
+        return ButtonDescription.newButtonDescription("diagram-filter/split-button/pin")
+                .idProvider(new WidgetIdProvider())
+                .targetObjectIdProvider(variableManager -> variableManager.get(VariableManager.SELF, Object.class).map(this.objectService::getId).orElse(null))
+                .labelProvider(variableManager -> "Pin")
+                .iconURLProvider(variableManager -> List.of())
+                .isReadOnlyProvider(variableManager -> false)
+                .buttonLabelProvider(variableManager -> "Pin")
+                .imageURLProvider(variableManager -> DiagramImageConstants.PIN_SVG)
+                .pushButtonHandler(variableManager -> {
+                    var diagram = variableManager.get("diagram", Diagram.class).get();
+                    var nodeIds = this.diagramFilterHelper.getSelectedElementIds(variableManager);
+                    var editingContext = variableManager.get("editingContext", IEditingContext.class).get();
+                    return this.diagramFilterHelper.sendDiagramEvent(variableManager, new PinDiagramElementInput(UUID.randomUUID(), editingContext.getId(), diagram.getId(), nodeIds, true));
+                })
+                .diagnosticsProvider(variableManager -> List.of())
+                .kindProvider(variableManager -> "")
+                .messageProvider(variableManager -> "")
+                .styleProvider(variableManager -> ButtonStyle.newButtonStyle()
+                        .backgroundColor("#ffffff")
+                        .foregroundColor("#000000")
+                        .build()
+                )
+                .helpTextProvider(variableManager -> "Pin selected elements")
```

Comment:
Use the internationalization support for text sent back to the end user (this applies to other button description provider)

Reply:
done